pandas/core/algorithms.py

```diff
@@ -413,35 +415,41 @@ def isin(comps, values):
         return comps._values.isin(values)
 
     comps = com._values_from_object(comps)
+    comps, dtype_comps, _ = _ensure_data(comps)
 
-    comps, dtype, _ = _ensure_data(comps)
-    values, _, _ = _ensure_data(values, dtype=dtype)
+    is_time_like = lambda x: (is_datetime_or_timedelta_dtype(x)
+                              or isinstance(x, Timestamp))
 
-    # faster for larger cases to use np.in1d
-    f = lambda x, y: htable.ismember_object(x, values)
+    is_int = lambda x: ((x == np.int64) or (x == int))
+
```

Comment:
woa, what are you doing? this is way less understandable that before. too many more if/thens here. pls fit this into the existing structure.

Reply:
Have edited to retain much of the existing structure

Reply:
Revised benchmarks after code restructure f8cc271 
```
       before           after         ratio
     [365eac4d]       [f8cc271b]
-     2.31±0.01ms      1.91±0.03ms     0.82  algorithms.Hashing.time_series_int

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
```

Reply:
Revised asv benchmarks on algorithms after 5416711
```
       before           after         ratio
     [365eac4d]       [54167114]
-     2.25±0.01ms      1.87±0.02ms     0.83  algorithms.Hashing.time_series_timedeltas

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
```

Reply:
@jreback thanks for the review - any other edits needed? 
Have tried to retain the existing structure and reduced if-blocks, though have removed the redundant try-except blocks. Have also tried to cluster the blocks logically.